ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtesthardware.py

```diff
@@ -0,0 +1,46 @@
+# Copyright (c) 2023 Arista Networks, Inc.
+# Use of this source code is governed by the Apache License 2.0
+# that can be found in the LICENSE file.
+from __future__ import annotations
+
+from functools import cached_property
+
+from ansible_collections.arista.avd.plugins.plugin_utils.eos_validate_state_utils.avdtestbase import AvdTestBase
+from ansible_collections.arista.avd.plugins.plugin_utils.utils import get
+
+
+class AvdTestHardware(AvdTestBase):
+    """
+    AvdTestHardware class for hardware tests.
+    """
+
+    anta_module = "anta.tests.hardware"
+    categories = ["Hardware"]
+
+    @cached_property
+    def test_definition(self) -> dict:
+        """
+        Generates the proper ANTA test definition for all hardware tests.
+
+        Returns:
+            test_definition (dict): ANTA test definition.
+        """
+        anta_tests = [
+            # Single test in ANTA covering 2 current eos_validate_state tests
+            {"VerifyEnvironmentCooling": {"states": get(self.task_vars, "validation_role.pwr_supply_states", ["ok"])}},
+            {
+                "VerifyEnvironmentPower": {
+                    "states": get(self.task_vars, "validation_role.fan_states", ["ok"]),
+                    "result_overwrite": {"categories": self.categories, "description": "Power supply status", "custom_field": "Power supply"},
+                }
```

Comment:
Looks like power supply and cooling tests have the wrong "states being picked up"